### ISSUE TITLE
TypeHintDeclaration.UselessDocComment: Object type variance missing

### ISSUE THREAD

**Issue by Majkl578** [slevomat/coding-standard#240]
There seems to be no variance support for `object` typehint.
For return types:
```php
/**
 * @return \stdClass
 */
public function foo() : object
{
}
```
and for parameter types:
```php
/**
 * @param \DateTimeImmutable|null $foo
 */
public function foo(?object $foo)
{
}
```

Both of these are reported as having useless comment: `Method Test::foo() does not need documentation comment.` But these are not useless - they specify a more specific type which can't be changed in signature itself due to invariance/LSP.

Happens with PHP 7.2.1RC1 and CS 4.3.0.

**Comment by kukulich** [comment]
Thank you for your report.

Fixed in https://github.com/slevomat/coding-standard/commit/823ba04bdcfe642ec2b912e4650d6e33d48c2ef7

**Comment by Majkl578** [comment]
Awesome, thanks!
